airbyte-integrations/connectors/source-gitlab/source_gitlab/streams.py

```diff
@@ -0,0 +1,338 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+import tempfile
+from abc import ABC
+from typing import Any, Dict, Iterable, List, Mapping, MutableMapping, Optional
+
+import requests
+import vcr
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.utils import casing
+
+cache_file = tempfile.NamedTemporaryFile()
+
+
+class GitlabStream(HttpStream, ABC):
+    primary_key = "id"
+    stream_base_params = {}
+    flatten_id_keys = []
+    flatten_list_keys = []
+    page = 1
+    per_page = 50
+
+    def __init__(self, api_url: str, **kwargs):
+        super().__init__(**kwargs)
+        self.api_url = api_url
+
+    @vcr.use_cassette(cache_file.name, record_mode="new_episodes", serializer="json")
+    def read_records(self, **kwargs) -> Iterable[Mapping[str, Any]]:
+        yield from super().read_records(**kwargs)
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        params = {"page": self.page, "per_page": self.per_page}
+        if next_page_token:
+            params.update(next_page_token)
+        params.update(self.stream_base_params)
+        return params
+
+    @property
+    def url_base(self) -> str:
+        return f"https://{self.api_url}//api/v4/"
```

Comment:
Why does it contain `//`? Is this a typo or do I miss smth?
<img width="552" alt="image" src="https://user-images.githubusercontent.com/19872253/122103734-9c81ea80-ce1f-11eb-8314-665fcbcac415.png">

Reply:
Yep, my typo, thanks